### PR TITLE
feat: ORA staff grader mfe endpoints upgraded

### PR DESCRIPTION
**Pull request by nandodev-net**
# ORA staff grader endpoints upgrade

For this PR we have the following goals:

-  Create the `/api/ora_staff_grader/assessments/feedback` endpoint in order to list the assessments grades based on the type (received or given) for a specific submission.
- Update the `/api/ora_staff_grader/initialize` endpoint in order to add the `fullname` and `email` fields.

# Descriptions

## /api/ora_staff_grader/assessments/feedback
This endpoint fetches assessment details for a given submission UUID, assessment_fillter [`given`, `received`] and ORA_location, including scorer information and scoring details in order to fill the following Staff Grade MFE table:
<p align="center">
  Descripción alternativa 1
</p>

Modifications were made in the following directories:

> #### edx-ora2
> - openassessment/staffgrader/staff_grader_mixin.py
> - openassessment/data.py
> #### edx-platform
> - lms/djangoapps/ora_staff_grader/constants.py
> - lms/djangoapps/ora_staff_grader/ora_api.py
> - lms/djangoapps/ora_staff_grader/serializers.py
> - lms/djangoapps/ora_staff_grader/urls.py
> - lms/djangoapps/ora_staff_grader/views.py

<details>
  <summary>Click here! to see `/api/ora_staff_grader/assessments/feedback` output</summary>

```json
  {
      "assessments": [
          {
              "id_assessment": "17",
              "scorer_name": "Fernando Gonzalez",
              "scorer_username": "admin",
              "scorer_email": "admin@admin.com",
              "assesment_date": "17-10-2023",
              "assesment_scores": [
                  {
                      "criterion_name": "Ideas",
                      "score_earned": 5,
                      "score_type": "Good"
                  },
                  {
                      "criterion_name": "Content",
                      "score_earned": 5,
                      "score_type": "Excellent"
                  }
              ],
              "problem_step": "Staff",
              "feedback": "OVERALL WENO"
          },
      ]
  }
```
</details>

&nbsp;

&nbsp;
## /api/ora_staff_grader/initialize
This is a pre-existing endpoint to which two extra fields are added [`fullname`, `email`].

<p align="center">
  Descripción alternativa 2
</p>

Modifications were made in the following directories:
> #### edx-ora2
> - /openassessment/staffgrader/tests/test_list_staff_workflows.py
> - openassessment/staffgrader/serializers/submission_list.py
> - openassessment/staffgrader/staff_grader_mixin.py
> - openassessment/data.py
> #### edx-platform
> - lms/djangoapps/ora_staff_grader/serializers.py

<details>
  <summary>Click here! to see `/api/ora_staff_grader/initialize` output</summary>

```json
...
...
    "submissions": {
        "619b8fb2-ebc0-49ed-9b35-4700efb28f66": {
            "submissionUUID": "619b8fb2-ebc0-49ed-9b35-4700efb28f66",
            "username": "admin",
            "email": "admin@admin.com",
            "fullname": "Fernando Gonzalez",
            "teamName": null,
            "dateSubmitted": "2023-10-13 15:36:13.781179+00:00",
            "dateGraded": "2023-10-17 20:26:07.146547+00:00",
            "gradedBy": "admin",
            "gradeStatus": "graded",
            "lockStatus": "unlocked",
            "score": {
                "pointsEarned": 10,
                "pointsPossible": 10
            }
        }
    },
```
</details>

&nbsp;

## Testing

In order to test this, you can use the API platform Postman:

1. Get an auth token from /oauth2/access_token/

- **/api/ora_staff_grader/initialize:** params: oraLocation
-  **/api/ora_staff_grader/assessments/feedback:** params: oraLocation, submissionUUID, assessmentFilter = [`given`, `received`]

&nbsp;

This upgrade will consist of two PRs: one for ORA and the other for edx-platform. This is because the main serializer of the service resides in the platform. To test it, you'll need to fetch and set up the version of the platform with the serializer modification.

- **Platform PR link**: https://github.com/openedx/edx-platform/pull/33600

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @nandodev-net!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: openassessment/staffgrader/tests/test_list_staff_workflows.py [hunk 1]
```diff
@@ -601,20 +623,18 @@ def test_bulk_fetch_annotated_staff_workflows(self, xblock, set_up_grades, set_u
...
             Mock(scorer_id=self.course_staff[1].student_id),
             Mock(assessment=None, scorer_id=None),
             Mock(assessment=None, scorer_id=None),
             Mock(scorer_id=self.course_staff[2].student_id),
         ]
+
         with self._mock_get_student_ids_by_submission_uuid() as mock_get_student_ids:
-            # with self._mock_get_team_ids_by_team_submission_uuid() as mock_get_team_ids:
-            with self._mock_map_anonymized_ids_to_usernames() as mock_map_ids:
+            with self._mock_map_anonymized_ids_to_user_data() as mock_map_data:  # Cambio importante aquí
```
Comment (BryanttV) [inline openassessment/staffgrader/tests/test_list_staff_workflows.py:635]: Can you remove the comment?

File: openassessment/staffgrader/tests/test_list_staff_workflows.py [hunk 2]
```diff
@@ -627,12 +647,14 @@ def test_get_list_workflows_serializer_context(self, xblock):
         expected_anonymous_id_lookups.update(
             {self.course_staff[1].student_id, self.course_staff[2].student_id}
         )
-        mock_map_ids.assert_called_once_with(expected_anonymous_id_lookups)
+        mock_map_data.assert_called_once_with(expected_anonymous_id_lookups)  # Cambio importante aquí
```
Comment (BryanttV) [inline openassessment/staffgrader/tests/test_list_staff_workflows.py:650]: Same as above

File: openassessment/staffgrader/tests/test_list_staff_workflows.py [hunk 3]
```diff
@@ -154,6 +154,28 @@ def _mock_map_anonymized_ids_to_usernames(self):
         ) as patched_map:
             yield patched_map
 
+    @contextmanager
+    def _mock_map_anonymized_ids_to_user_data(self):
+        """
+        Context manager that patches map_anonymized_ids_to_user_data and returns a mapping from student IDs to a dictionary
+        containing username, email, and fullname.
+        """
+        # Creamos un diccionario simulado que represente lo que la función real devuelve
```
Comment (BryanttV) [inline openassessment/staffgrader/tests/test_list_staff_workflows.py:163]: Same as above

File: openassessment/staffgrader/staff_grader_mixin.py [hunk 4]
```diff
@@ -193,6 +199,42 @@ def list_staff_workflows(self, data, suffix=''):  # pylint: disable=unused-argum
...
+            list[dict]: A list of dictionaries, each representing an assessment's data.
+
+        Note:
+            - If 'assessment_filter' is "received", the function fetches assessments received
+            for the given 'submission_uuid'.
+            - For any other value of 'assessment_filter', the function fetches assessments
+            given by the owner of the 'submission_uuid' for other submissions in the same item.
+        """
+        item_id = data['item_id']
+        subission_uuid = data['submission_uuid']
```
Comment (BryanttV) [inline openassessment/staffgrader/staff_grader_mixin.py:227]:
```suggestion
        submission_uuid = data['submission_uuid']
```

File: openassessment/data.py [hunk 5]
```diff
@@ -1535,3 +1565,161 @@ def get_file_uploads(self, missing_blank=False):
...
+    for assessment in assessment_list:
+
+        scorer_name, scorer_username, scorer_email = get_scorer_data(assessment.scorer_id, user_data_mapping)
+
+        assessment_data_list.append({
+            "assessment_id": str(assessment.id),
+            "scorer_name": scorer_name,
+            "scorer_username": scorer_username,
+            "scorer_email": scorer_email,
+            "assesment_date": assessment.scored_at.strftime('%d-%m-%Y'),
```
Comment (BryanttV) [inline openassessment/data.py:1659]: Shouldn't the time return as well?

**Comment by BryanttV** [comment]
This work done by @nandodev-net will be continued from the next [PR](https://github.com/openedx/edx-ora2/pull/2101). From now on, I will take responsibility for any modifications that need to be made in the implementation.

**Comment by openedx-webhooks** [comment]
@nandodev-net Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2095&entry.1484655318=https://github.com/nandodev-net&entry.752974735=2023-10-25+23:40&entry.1917517419=2023-11-01+18:42&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**